### PR TITLE
Updated API priority and fairness wrt defaulting behaviors plus related changes

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change

/kind bug

> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
This builds on #81834 by adding a commit that fixes divergences from the desired behavior described in https://github.com/kubernetes/enhancements/pull/1221 .  The only thing new in this PR is the last commit.

Simplified involved code.

Factored NewRequestManager to expose the testable kernel of its
functionality.

Revised some names following suggestions from Daniel Smith review, as
they were involved in the same code.  Following the suggestion to
simplify "NewRequestManagementSystem" to "NewRequestManager", renamed
the associated private types: requestManagementSystem ->
requestManager, requestManagementState -> requestManagerState.
Following the objection to the vague term "request management"
replaced it with "API priority and fairness" where that worked easily,
"flow control" in other places.  Kept "request manage..." only where
replacing it with "flow control" would stutter.  But did not change
the feature gate name.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
(actually a release note is coming, but is not in this PR)

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
[KEP] https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md
```

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
@MikeSpreitzer

File: staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt.go [hunk 1]
```diff
@@ -120,72 +122,81 @@ type requestManagementSystem struct {
...
 }
 
-// NewRequestManagerWithPreservation creates a new instance
-// of request-management system with preservation.  The WaitGroup is
-// optional and, if supplied, is kept informed of whenever a goroutine
-// is started or stopped or begins or finishes waiting --- except that
-// the configuration controller is not fully plumbed yet.
-func NewRequestManagerWithPreservation(
+// NewRequestManagerTestable is extra flexible to facilitate testing
+func NewRequestManagerTestable(
```
Comment (yue9944882) [inline staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt.go]: moving this to `reqmgmt_test.go?`

**Pull request closed without merging**